arangod/VocBase/LogicalCollection.cpp

```diff
@@ -1212,6 +1217,14 @@ void LogicalCollection::decorateWithInternalValidators() {
   decorateWithInternalEEValidators();
 }
 
+replication2::LogId LogicalCollection::shardIdToLogId(ShardID const& shardId) {
+  auto stateId = std::string_view(shardId).substr(1, shardId.size() - 1);
+  auto logId = replication2::LogId::fromString(stateId);
+  ADB_PROD_ASSERT(logId.has_value())
+      << " converting " << shardId << " to LogId failed";
+  return logId.value();
+}
```

Comment:
Maybe we should move this into a static `fromShardId` function in `LogId` (similar to `fromString`).

Reply:
I did it like that initially, but on second thought I considered it would be too much to add knowledge about shards inside the `LogId` class. I figured that a `LogicalCollection` is the place where such a connection between the two takes place. This is debatable and I understand your point, it could make the method a bit more apparent to everyone looking for it.

Reply:
What do you think?

Reply:
Makes sense!